### PR TITLE
test: writing the fixtures, rewriting tests

### PR DESCRIPTION
**Pull request by CTAPCKPIM**
- `fixtures` can be added as tests are written

### CODE REVIEW

**Comment by zolotokrylin** [comment]
How it comes PR is open 2 hours ago, but we see 2 weeks old commits?

**Comment by CTAPCKPIM** [comment]
@zolotokrylin 

> How it comes PR is open 2 hours ago, but we see 2 weeks old commits?

I don’t understand, in fact, I perform all tasks from one local folder, and maybe that’s why it affects so much that you can see the old commits, but I don’t understand what I should do

**Comment by CTAPCKPIM** [comment]
- `StreamManager.test.js` will delete after the review of my **PR**

File: test/StreamManagerTest.js [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+		})
+	})
+
+	describe("changePayerAddress();", async () => {
+		// Tests for `changePayerAddress();`
+		// Changing the address payer
+		it('Change address payer: address of the admin is changing', async () => {
+
+			const { admin, payer, payee1, payee2 } = await loadFixture(getSigners)
+	    	const { streamManager, mockUSDT } = await loadFixture(getDeployContracts)
```
Comment (georgeciubotaru) [inline test/StreamManagerTest.js:78]: @CTAPCKPIM Just combine these 2 fixtures into 1

**Comment by zolotokrylin** [comment]
Problem: Daniil's PRs include old irrelevant commits

File: test/StreamManagerTest.js [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+    const terminationPeriod = 18 * 24 * 3600; // 18 days
+    const cliffPeriod = 24 * 3600; // 24 hrs
+    const zero = ethers.constants.AddressZero 
+
+	it("Checking fixtures", async () => {
+		await loadFixture(getSignersAndDeployContracts)
+
+	    // Returning the `decimals` for coverage*
+	    const { mockUSDT } = await loadFixture(getSignersAndDeployContracts)
+	    expect(await mockUSDT.decimals()).to.eq(6)
```
Comment (georgeciubotaru) [inline test/StreamManagerTest.js:18]: @CTAPCKPIM The idea here is the following..

Fixture 1 (deploy the contracts)
Fixture 2 (creating something)
Fixture 3 (removing something)

if you need to use Fixture 3, it should contain the first 2, and so on.

In your case I see here duplication.

File: test/StreamManagerTest.js [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+  	})
+	
+	describe("changeCommissionAddress();", async () => {
+		// Tests for `changeCommissionAddress();`
+		// Changing the address of the commission
+		it('Chainge address fee: address of the admin is changing', async () => {
+	    	
+	    	const { admin, payee1, streamManager } = await loadFixture(getSignersAndDeployContracts)
```
Comment (georgeciubotaru) [inline test/StreamManagerTest.js:26]: @CTAPCKPIM I like your style of writing tests.

Try to write tests starting from:

How the test cases are declared in the function

1. First modifiers
2. Then one by one the revert statement
3. Success case
4. Case for second trigger (if it's available)

File: test/StreamManagerTest.js [hunk 4]
```diff
@@ -0,0 +1,769 @@
...
+		    .to.be.revertedWith('NotPayer');
+		})
+	})
+
+	describe("terminate();", async () => {
+		// Tests for `terminate();`
+		// Expecting success
+		it('Terminate: terminating succeed;', async () => {
+
+	    	const { payer, payee1, streamManager, mockUSDT } = await loadFixture(getSignersAndDeployContracts)
```
Comment (georgeciubotaru) [inline test/StreamManagerTest.js:180]: here you can have a fixture with an open stream created and use for all cases

File: test/StreamManagerTest.js [hunk 5]
```diff
@@ -0,0 +1,769 @@
...
+		    expect(await streamManager.isPayee(payee1.address)).to.equal(false)
+		  })
+	})
+
+	describe("accumulation();", async () => {
+		// Tests for `accumulation();`
+		// Amount is accumulated
+		it('Accumulation: returning the amount;', async () => {
+
+	    	const { payer, payee1, streamManager, mockUSDT } = await loadFixture(getSignersAndDeployContracts)
```
Comment (georgeciubotaru) [inline test/StreamManagerTest.js:263]: same here. try to have a DRY code. Thank you

**Comment by georgeciubotaru** [comment]
@CTAPCKPIM Let's try this way. You merge this PR and create a new PR for each `function` with the all test cases contained.

Your PR should not be longer than 4hrs

**Pull request merged**